lib/extensions/notify_delivery_method.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Mail
+  module Notify
+    class Personalisation
+      BLANK = Object.new
+
+      module BlankFix
+        def to_h
+          super.transform_values do |value|
+            value == BLANK ? "" : value
+          end
+        end
+      end
```

Comment:
I am not quite sure why we need a constant for blank. The reasonable behaviour, to me, is that when you pass something empty or falsy as a value in personalisation, it gets to be an empty string when sent to notify. 

So something like 
```
      def to_h
        merged_options.map { |_k, v| v.blank? ? "" : v }
      end
```

Thoughts?

Reply:
I definitely prefer `fail-by-default` here. Yes, it is up for us to handle our blanks, which is why we need to mark those personalisation as potentially blank if we really want them to be blank. All the other blank has to fail, as otherwise we're sending incomplete emails.

There are not gona be a lot of places to do that mapping either - this would only happen in mailer as it is closely related to the structure of the template. I've added `blank_allowed` method to make it easier and to encapsulate the existence of `BLANK` constant. I had to do manual mapping in the Redirector as this happens outside of the mailer, but I don't think there are any other places we'd need to do this.